syncthing.te

```diff
@@ -0,0 +1,90 @@
+policy_module(syncthing, 1.0.0)
+
+# Declarations
+attribute_role syncthing_roles;
+role syncthing_roles types syncthing_t;
+
+type syncthing_t;
+type syncthing_exec_t;
+# should only ever be execed non-root, but can be started by systemd --user as a service
+init_daemon_domain(syncthing_t, syncthing_exec_t)
+userdom_user_application_domain(syncthing_t, syncthing_exec_t)
+userdom_use_user_ptys(syncthing_t)
+
+type syncthing_config_home_t;
+userdom_user_home_content(syncthing_config_home_t)
+
+# Local Policy Rules
+
+# file permissions
+fs_getattr_xattr_fs(syncthing_t)
+
+# manage user home content files/dirs/symlinks
+userdom_manage_user_home_content_files(syncthing_t)
+userdom_manage_user_home_content_dirs(syncthing_t)
+userdom_manage_user_home_content_symlinks(syncthing_t)
+
+userdom_delete_user_home_content_files(syncthing_t)
+userdom_delete_user_home_content_dirs(syncthing_t)
+userdom_delete_user_home_content_symlinks(syncthing_t)
+
+userdom_read_user_home_content_files(syncthing_t)
+userdom_read_user_home_content_symlinks(syncthing_t)
+
+# newly created dirs in home root will transition to user_home_t
+userdom_user_home_dir_filetrans_user_home_content(syncthing_t, dir)
+
+# self rules
+allow syncthing_t self:process getsched;
+allow syncthing_t self:fifo_file { read write };
+
+# read randomness
+dev_read_rand(syncthing_t)
+dev_read_urand(syncthing_t)
+
+# networking
+allow syncthing_t self:tcp_socket create_stream_socket_perms;
+allow syncthing_t self:udp_socket create_socket_perms;
+
+allow syncthing_t syncthing_exec_t:file execute_no_trans; # presumably allowed to restart itself
+
+corenet_tcp_sendrecv_generic_if(syncthing_t)
+corenet_udp_sendrecv_generic_if(syncthing_t)
+
+corenet_tcp_bind_generic_node(syncthing_t)
+corenet_tcp_sendrecv_generic_node(syncthing_t)
+
+corenet_udp_bind_generic_node(syncthing_t)
+corenet_udp_sendrecv_generic_node(syncthing_t)
+
+# apparently some relays are running on port 990?
+corenet_tcp_connect_all_ports(syncthing_t)
+
+corenet_tcp_sendrecv_all_ports(syncthing_t)
+corenet_udp_sendrecv_all_ports(syncthing_t)
+
+corenet_tcp_bind_syncthing_port(syncthing_t)
+corenet_udp_bind_syncthing_discovery_port(syncthing_t)
+corenet_tcp_bind_syncthing_admin_port(syncthing_t)
+
+miscfiles_read_generic_certs(syncthing_t) # read TLS certificates
+miscfiles_read_localization(syncthing_t) # read timezone information
+
+kernel_read_kernel_sysctls(syncthing_t)
+
+kernel_read_system_state(syncthing_t)
+kernel_read_net_sysctls(syncthing_t)
+
+sysnet_dns_name_resolve(syncthing_t)
+auth_use_nsswitch(syncthing_t) # /etc/nsswitch.conf
+
+networkmanager_read_pid_files(syncthing_t)
+
+# TODO no rule allows searching this dir type
+gen_require(`
+    type NetworkManager_var_run_t;
+')
+search_dirs_pattern(syncthing_t, NetworkManager_var_run_t, NetworkManager_var_run_t)
```

Comment:
line 83 to 87 can be removed (is already allowed with the networkmanager_read_pid_files() call